kernel/src/grant.rs

```diff
@@ -90,9 +90,11 @@ impl Allocator {
                     process
                         .alloc(size_of::<T>())
                         .map_or(Err(Error::OutOfMemory), |arr| {
-                            let mut owned = Owned::new(arr.as_mut_ptr() as *mut T, self.appid);
-                            *owned = data;
-                            Ok(owned)
+                            let ptr = arr.as_mut_ptr() as *mut T;
+                            // We use `ptr::write` to avoid `Drop`ping the uninitialized memory in
+                            // case `T` implements the `Drop` trait.
+                            write(ptr, data);
```

Comment:
why are you not using ```mem::forget``` I thought it was more or less implemented for this use-case?

Reply:
`mem::forget` takes ownership of the value (and thus `ptr` would be inaccessible), which is not what we want. We want to _replace_ the value (which is uninitialized memory) with a new one, without the compiler thinking we want to `drop` what was sitting in `ptr` before.